src/hooks.ts

```diff
@@ -265,24 +268,35 @@ export const useFeature: UseFeature = (featureKey, options = {}, overrides = {})
   const [prevDecisionInputs, setPrevDecisionInputs] = useState<DecisionInputs>(currentDecisionInputs);
   if (!areDecisionInputsEqual(prevDecisionInputs, currentDecisionInputs)) {
     setPrevDecisionInputs(currentDecisionInputs);
-    setDecisionState(getCurrentDecision());
+    setState(prevState => ({
+      ...prevState,
+      ...getCurrentDecision(),
+    }));
   }
 
+  const finalReadyTimeout = options.timeout !== undefined ? options.timeout : timeout;
+  useEffect(() => {
+    if (!isClientReady) {
+      subscribeToInitialization(optimizely, finalReadyTimeout, initState => {
+        setState({
+          ...getCurrentDecision(),
+          ...initState,
+        });
+      });
+    }
+  }, [isClientReady, finalReadyTimeout, setState, getCurrentDecision, optimizely]);
+
   useEffect(() => {
-    if (!isClientReady || options.autoUpdate) {
+    if (options.autoUpdate) {
       return setupAutoUpdateListeners(optimizely, HookType.FEATURE, featureKey, hooksLogger, () => {
-        setDecisionState(getCurrentDecision());
+        setState(prevState => ({
+          ...prevState,
+          ...getCurrentDecision(),
+        }));
       });
     }
     return (): void => {};
```

Comment:
more of a question on the previously existing functionality - does this do anything helpful or could it be left off? No need to change that here, jw. My understanding is it's optional and isn't helpful here.

Reply:
Ohh, I see - it's because we're returning the cancelation up above so probably have to always supply it.